src/data/repositories/UserD2ApiRepository.ts

```diff
@@ -152,6 +152,22 @@ export class UserD2ApiRepository implements UserRepository {
         ).map(({ objects }) => objects.map(user => user.id));
     }
 
+    public listAllUsernames(options: ListOptions): FutureData<string[]> {
```

Comment:
A repository should never return primitive types, this is a code smell that we saw in the coding dojo

In Clean Architecture, a repository should always return entities.

The presentation layer should map user list to get userNames if required.

In this case the entity is UserLogic (can we change to User and User to UserProps?)